pyproject.toml

```diff
@@ -16,28 +16,19 @@ omegaconf = "^2.1.1"
 # Why do we modify the requirements?
 # This framework is intended for running common deep learning models for point cloud analysis.
 # The original requirements are quite diverse and are not suited for our purposes (i.e. running KPConv in OSS).
-# Hence, we strip out some of the dependencies that are not needed for our use case.
-# We also modify a few dependencies to provide direct links to wheels for faster installation.
-
-# Of course, being too specific with the requirements is not ideal, as we will need to update the requirements
-# when we update the framework OR when we change the dependencies such as torch-1.10.0-cu113 or Python version (currently 3.8).
-# That means, we need to update the requirements when we update the pytorch version OR when we change cuda version in OSS.
-
-# We set the versions for torch-{cluster, scatter, sparse} fixed here because you need torch installed before poetry can build these.
-# See https://github.com/rusty1s/pytorch_scatter/issues/252. We have to use specific wheels instead.
 
 # Another thing to note is that torch-cluster, torch-scatter and torch-sparse wheels are
-# provided by pyg.org, while torch-points-kernels whell is provided by promaton.
-# Please see OSS README for more details on how to build the wheels.
+# provided by pyg.org, while torch-points-kernels whell is provided by Promaton.
+# Please see https://github.com/promaton/ios/tree/main/projects/oss#third-party-libraries for more details on how to build the wheels.
 
-torch-points-kernels = { url= "https://pypi.dev.promaton.ai/packages/torch_points_kernels-0.7.1+cu117-cp310-cp310-linux_x86_64.whl" }
-torch-cluster = {url = "https://data.pyg.org/whl/torch-2.0.0%2Bcu117/torch_cluster-1.6.3%2Bpt20cu117-cp310-cp310-linux_x86_64.whl"}
-torch-scatter = {url = "https://data.pyg.org/whl/torch-2.0.0%2Bcu117/torch_scatter-2.1.2%2Bpt20cu117-cp310-cp310-linux_x86_64.whl"}
-torch-sparse = {url = "https://data.pyg.org/whl/torch-2.0.0%2Bcu117/torch_sparse-0.6.18%2Bpt20cu117-cp310-cp310-linux_x86_64.whl"}
-torch-geometric = "^1.7.2"
+torch-points-kernels = { version = "==0.7.2+cu121", source = "promaton" }
+torch-cluster = { version = "==1.6.3+pt24cu121", source = "pyg" }
+torch-scatter = { version = "==2.1.2+pt24cu121", source = "pyg" }
+torch-sparse = { version = "==0.6.18+pt24cu121", source = "pyg" }
+torch-geometric = "^2.6.1"
 
 # Additional constraints on no major version creep
-torch = "~2.0.1"
+torch = "~2.4.1"
```

Comment:
Is this compatible with torch `2.0.1` currently used e.g., in acd?

Reply:
Actually, this PR and the corresponding published wheel targets the Torch 2.4.1 upgrade PR on the IOS side - https://github.com/promaton/ios/pull/2953.

So, they are compatible yes.